### PR TITLE
Search on fee_type, amount, and receipt json

### PR DESCRIPTION
**Pull request by nessur**
## 📚 Purpose
*[adds full-text search to receipt json, range filtering on amount, matching on fee_type]*

## 👌 Resolves:
- [ ] ✨ SS-20 adds awesome transactions-api features

## 📦 Impacts:
*[mds-transactions-service]*

## ✅ PR Checklist
- [x] Update RDS to PG11

### CODE REVIEW

File: packages/mds-transaction-service/tests/transactions.spec.ts [hunk 1]
```diff
@@ -258,6 +260,93 @@ describe('Transaction Service Tests', () => {
...
+          })
+          expect(redPages.length).toEqual(0) // no results are base_fee
+
+          // Arbitrarily generate 5 events
+          await TransactionServiceClient.createTransactions([
+            ...transactionsGenerator(5, { provider_id, receipt_details: { color: 'Red' }, amount: 125 })
+          ])
+
+          // Partial matching on string-value 'red', amount > 100 and amount < 200
+          const { transactions: whitePages } = await TransactionServiceClient.getTransactions({
```
Comment (janedotx) [inline packages/mds-transaction-service/tests/transactions.spec.ts:333]: Can we change `whitePages` to `redPages`?

File: packages/mds-transaction-service/tests/transactions.spec.ts [hunk 2]
```diff
@@ -258,6 +260,93 @@ describe('Transaction Service Tests', () => {
...
+          // Partial matching on string-value 'blu', fee_type is parking_fee
+          const { transactions: bluePages } = await TransactionServiceClient.getTransactions({
+            limit,
+            search_text: 'blu',
+            fee_type: 'parking_fee'
+          })
+          expect(bluePages.length).toEqual(limit) // all results are color:blue
+
+          // Search for a non-existent value
+          const { transactions: redPages } = await TransactionServiceClient.getTransactions({
```
Comment (janedotx) [inline packages/mds-transaction-service/tests/transactions.spec.ts:320]: Can we change `redPages` to `baseFeePages`?

**Review by janedotx — APPROVED** [review]
LGTM pending some nits

**Review by cjlynch12 — APPROVED** [review]

**Review by avatarneil — APPROVED** [review]
Same nits as @janedotx, but LGTM

File: packages/mds-transaction-service/tests/transactions.spec.ts (hunk 2 above)
Comment (nessur) [reply packages/mds-transaction-service/tests/transactions.spec.ts]: fixy!

File: packages/mds-transaction-service/tests/transactions.spec.ts (hunk 1 above)
Comment (nessur) [reply packages/mds-transaction-service/tests/transactions.spec.ts]: fixed!

**Pull request merged**